textx/model.py

```diff
@@ -693,7 +743,12 @@ def call_obj_processors(metamodel, model_obj,
         # enter recursive visit of attributes only, if the class of the
         # object being processed is a meta class of the current meta model
         if model_obj.__class__.__name__ in metamodel:
-            current_metaclass_of_obj = metamodel[model_obj.__class__.__name__]
+            if hasattr(model_obj, '_tx_fqn'):
+                current_metaclass_of_obj = metamodel[model_obj._tx_fqn]
+            else:
+                # fallback (not used - unsure if this case is required...):
+                current_metaclass_of_obj = metamodel[model_obj.__class__.__name__]
+            assert current_metaclass_of_obj is not None
```

Comment:
This code block should be reviewed carefully. I had issues with grammars where rules are redefined (`test_user_class_with_imported_grammar` in https://github.com/textX/textX/blob/master/tests/functional/test_user_classes.py#L94) during code changes. I did never run into the fallback case - and (since we always provide a `_tx_fqn` field) we could probably remove the else case...

Reply:
Yes, this might be a problem as the `if` condition is using a plain name as a guard to enter this block. I haven't investigated this in greater details as I want to provide the first feedback ASAP.

Reply:
@igordejanovic : Did you had time to investigate this point? If not we can either chose to (a) make more tests (I could do this the next time). Or (b) we can first merge and make this investigations separately (I think: if we find effects here, these effects are independent of this PR - and are just made visible by it...).

My favorite is to make the investigations separately... (to reduce the clutter induced by this PR getting too large).. What do you think?

Reply:
I agree. We can investigate that separately.